### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by gsmalik030**
In this PR I completed the following tasks:

- [x] Created a class Classroom with a label instance variable.
- [x] Setter and getter for label.
- [x] Created the has-many/belongs-to relationship between Classroom and Student.
- [x] Created a class Book with title and author instance variables.
- [x] Setters and getters for instance variables.
- [x] created a class Rental with a date instance variable.
- [x] Setter and getter for date.
- [x] created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.
- [x] Fixed RoboCup errors.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require_relative 'nameable'
 
 class Person < Nameable
```
Comment (TedLivist) [inline person.rb:3]: - [ ] Your Person class is missing the method that establishes the relationship between itself and book through rentals. Kindly add that. See an example below:

  ```
  def add_rental(date, book)
    Rental.new(date, self, book)
  end
  ```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (TedLivist) [inline book.rb]: - [ ] Your Book class is missing the method that establishes the relationship between itself and person through rentals. Kindly add that. See an example below:

  ```
  def add_rental(date, person)
    Rental.new(date, person, self)
  end
  ```

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(lable)
+    @lable = lable
+    @students = []
+  end
+
+  def add_students(student)
+    @student.push(student)
+    student.update_classroom(self)
+  end
```
Comment (TedLivist) [inline classroom.rb]: - [ ] On line 10 we should have `@students`, not `student` as that is the array you defined.

- [ ] I would recommend you make this method independent of the one in the Student class by setting the classroom of the passed student object here.

  ```
  def add_student(student)
    @students.push(student)
     student.classroom = self
  end
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @gsmalik030,

Good job so far!
✔️ Association between student and classroom
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @gsmalik030, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :green_circle: 

- PR is professional :+1: 
- Classes have been associated :+1: 


## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**